### ISSUE TITLE
[HTML placeholders for asset injection] Improve DX, performance, and docs

### ISSUE THREAD

**Issue by brillout** [brillout/vite-plugin-ssr#695]
### Description

`vite-plugin-ssr@0.4.92` released placeholders to control where assets are injected in the HTML:

```html
<html>
  <head>
     <meta charset="UTF-8">
     __VITE_PLUGIN_SSR__ASSETS_FIRST__
  </head>
  <body>
     <!-- ... -->
     __VITE_PLUGIN_SSR__ASSETS_LAST__
  </body>
</html>
```

**TODO 1 - Improve DX**

```js
import { html } from 'vite-plugin-ssr'

export function onRenderHtml(pageContext) {
  const pageHtml = /* ... */
  return html`<!DOCTYPE html>
    <html>
      <head>
         <meta charset="UTF-8">
         ${html.assetsFirst}
      </head>
      <body>
        <div id="page-root">${html.dangerouslySkipEscape(pageHtml)}</div>
        ${html.assetsLast}
      </body>
    </html>`
}
```

**TODO 2 - Mention this in the docs**

**TODO 3 - Improve performance**

We can and should inject assets without touching `pageHtml` which can be potentially very big.

**Comment by chen6516** [comment]
> ### Description
> `vite-plugin-ssr@0.4.92` released placeholders to control where assets are injected in the HTML:
> 
> ```
> <html>
>   <head>
>      <meta charset="UTF-8">
>      __VITE_PLUGIN_SSR__ASSETS_FIRST__
>   </head>
>   <body>
>      
>      __VITE_PLUGIN_SSR__ASSETS_LAST__
>   </body>
> </html>
> ```
> 
> **TODO 1 - Improve DX**
> 
> ```js
> import { html } from 'vite-plugin-ssr/server'
> 
> export function onRenderHtml(pageContext) {
>   const pageHtml = /* ... */
>   return html`<!DOCTYPE html>
>     <html>
>       <head>
>          <meta charset="UTF-8">
>          ${html.assetsFirst}
>       </head>
>       <body>
>         <div id="page-root">${html.dangerouslySkipEscape(pageHtml)}</div>
>         ${html.assetsLast}
>       </body>
>     </html>`
> }
> ```
> 
> **TODO 2 - Mention this in the docs**
> 
> **TODO 3 - Improve performance**
> 
> We can and should inject assets without touching `pageHtml` which can be potentially very big.

as
 `__VITE_PLUGIN_SSR__ASSETS_LAST__` 

to be
`__VITE_PLUGIN__SSR_ASSETS_LAST__`

**Comment by brillout** [comment]
@chen6516 I'm not sure I see the added value here. As a user, you won't directly use these.

**Comment by FlorianCassayre** [comment]
> This will also enable placing `<meta charset="UTF-8" />` within the first 1024 bytes of HTML, see [vikejs/vike-vue#177](https://github.com/vikejs/vike-vue/issues/177).

This is highly desirable IMO.

For instance, the [Mattermost opengraph parser](https://github.com/mattermost/mattermost/blob/9ad0f1e0701e984b3fb83757f12728bc4a24e211/server/channels/app/opengraph.go#L79) falls back to `CP-1252` on my [website](https://florian.cassayre.me), because of the underlying [golang standard library](https://github.com/golang/net/blob/master/html/charset/charset.go#L53-L55) implementation.

I see that you are working on it, thanks!

**Comment by brillout** [comment]
Indeed, it's significant. Let's see if there are also other significant reasons and let's bump the priority of this then. (So far it still isn't a high priority on my radar, but I'm open to being convinced.)
